# Thaw-Ponds

## Bioinformatic tools for nitrogenase gene mapping and abundance estimations per sample

 To calculate the abundances of nitrogenase genes per sample, we worked with un-assemble sequencing data.  
 1) Raw reads were trimmed using Sickle
 2) To pair the reads, forward and reverse reads were merged using PEAR. 
 3) After merging the reads, the average length was 150 bp.
 4) Above is detailed quality information according to prinseq output
 
 Table 1. Sequence average per sample:
 
     Sample                              Average sequence length(bp)            Standar dev.(bp)
       A1                                         155.55                         ±49.22
       A2                                         150.72                         ±49.72
       A3                                         153.83                         ±49.22
       A4                                         153.02                         ±49.75
       B1                                         151.66                         ±49.37
       B2                                         152.82                         ±49.11 
       B4                                         154.23                         ±48.73 
       C2                                         154.51                         ±48.54 
       C3                                         151.22                         ±49.10
       C4                                         144.02                         ±48.72 
       C5                                         147.01                         ±49.52 
       D1                                         151.45                         ±48.80
       D2                                         147.98                         ±49.32
       D3                                         146.48                         ±48.92  
       
       


Detailed graphical information and the reports of quality control per sample in the links below:

  [A1 graphic](./png_graphs.zip)  and   [A1 report](./A1.doc)
  
  [A2 graphic](./png_graphs_A2.zip) and  [A2 report](./A2.doc)

  [A3 graphic](./png_graphs_A3.zip) and  [A3 report](./A3.doc)
  
  [A4 graphic](./png_graphs_A4.zip) and  [A4 report](./A4.doc)
  
  [B1 graphic](./png_graphs_B1.zip) and  [B1 report](./B1.doc)
  
  [B2 graphic](./png_graphs_B2.zip) and  [B2 report](./B2.doc)
  
  [B4 graphic](./png_graphs_B4.zip) and  [B4 report](./B4.doc)
  
  [C2 graphic](./png_graphs_C2.zip) and  [C2 report](./C2.doc)
  
  [C3 graphic](./png_graphs_C3.zip) and  [C3 report](./C3.doc)
  
  [C4 graphic](./png_graphs_C4.zip) and  [C4 report](./C4.doc)
  
  [C5 graphic](./png_graphs_C5.zip) and  [C5 report](./C5.doc)
  
  [D1 graphic](./png_graphs_D1.zip) and  [D1 report](./D1.doc)
  
  [D2 graphic](./png_graphs_D2.zip) and  [D2 report](./D2.doc)
  
  [D3 graphic](./png_graphs_D3.zip) and  [D3 report](./D3.doc)
  
  [D4 graphic](./png_graphs_D4.zip) and  [D4 report](./D4.doc)
  
  [C5 graphic](./png_graphs_C5.zip) and  [C5 report](./C5.doc)
  
  [C5 graphic](./png_graphs_C5.zip) and  [C5 report](./C5.doc)
